Modules/timemodule.c

```diff
@@ -559,26 +559,48 @@ See the library reference manual for formatting codes (same as strftime()).");
 
 
 static PyObject *
+_asctime(struct tm *timeptr)
+{
+    /* Inspired by Open Group reference implementation available at
+     * http://pubs.opengroup.org/onlinepubs/009695399/functions/asctime.html */
+    static const char wday_name[7][4] = {
+        "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"
+    };
+    static const char mon_name[12][4] = {
+        "Jan", "Feb", "Mar", "Apr", "May", "Jun",
+        "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"
+    };
+    PyObject *unicode, *str;
+    /* PyString_FromString() cannot be used because it doesn't support %3d */
+    unicode = PyUnicode_FromFormat(
+        "%s %s%3d %.2d:%.2d:%.2d %d",
+        wday_name[timeptr->tm_wday],
+        mon_name[timeptr->tm_mon],
```

Comment:
The function should verify that ``tm_wday`` and ``tm_mon`` are not under- or overflowing.

Reply:
Such check is done in the caller:

```
>>> time.asctime((2017, 13, 4, 22, 44, 18, 0, 247, 1))
ValueError: month out of range
```

Reply:
Ooops, sorry, I was checking Python 3.

You are right that my PR is wrong:

```
>>> time.asctime((2017, 13, 4, 22, 44, 18, 0, 247, 1))
'Mon ???  4 22:44:18 2017'
```

Reply:
It's now fixed:

```
>>> time.asctime((2017, 13, 4, 22, 44, 18, 0, 247, 1))
ValueError: month out of range
```